src/Query/Query.php

```diff
@@ -708,7 +709,21 @@ public function union(QueryInterface|string $sql, bool $all = false): static
         return $this;
     }
 
+    /**
+     * @throws LogicException
```

Comment:
Add description of exception and move this phpdoc to `QueryPartsInterface`. I think it should be part of interface.